sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/inprocess/BoundedReadEvaluatorFactory.java

```diff
@@ -104,18 +103,13 @@
   private static class BoundedReadEvaluator<OutputT> implements TransformEvaluator<Object> {
```

Comment:
It might be worth a smidgin of implementation detail javadoc here mentioning that the reader is only actually opened on finish bundle. It is a private class anyhow, but may direct your future self to the right place to look if there is a bug.